### ISSUE TITLE
Migration issues when updating from Mautic 4.9.11 to 5.x

### ISSUE THREAD

**Issue by wsarisslz** [mautic/mautic#13281]
### Mautic Version

5.0.x series

### PHP version

8.1

### What browsers are you seeing the problem on?

Not relevant

### What happened?

When updating in console to mautic 5.x the apply --finish command fails at the database update.

I also tried:
/var/www/mautic# sudo -u www-data php8.1 /var/www/mautic/bin/console doctrine:migration:migrate

 WARNING! You are about to execute a migration in database "mautic" that could result in schema changes and data loss. Are you sure you wish to continue? (yes/no) [yes]:
 > yes

                                                                                                                        
 [WARNING] You have 49 previously executed migrations in the database that are not registered migrations.               
                                                                                                                        
Are you sure you wish to continue? (yes/no) [yes]:
 > yes

```
[notice] Migrating up to Mautic\Migrations\Versionzz20230929183000
[error] Migration Mautic\Migrations\Version20200415135706 failed during Pre-Checks. Error: "Unknown column type "json_array" requested. Any Doctrine type that you use has to be registered with \Doctrine\DBAL\Types\Type::addType(). You can get a list of all the known types with \Doctrine\DBAL\Types\Type::getTypesMap(). If this error occurs during database introspection then you might have forgotten to register all database types for a Doctrine Type. Use AbstractPlatform#registerDoctrineTypeMapping() or have your custom types implement Type#getMappedDatabaseTypes(). If the type name is empty you might have a problem with the cache or forgot some mapping information."

In Exception.php line 115:
                                                                                                                                                                                              
  Unknown column type "json_array" requested. Any Doctrine type that you use has to be registered with \Doctrine\DBAL\Types\Type::addType(). You can get a list of all the known types with   
  \Doctrine\DBAL\Types\Type::getTypesMap(). If this error occurs during database introspection then you might have forgotten to register all database types for a Doctrine Type. Use Abstrac  
  tPlatform#registerDoctrineTypeMapping() or have your custom types implement Type#getMappedDatabaseTypes(). If the type name is empty you might have a problem with the cache or forgot som  
  e mapping information.                                                                                                                                                                      
                                                                                                                                                                                              

doctrine:migrations:migrate [--write-sql [WRITE-SQL]] [--dry-run] [--query-time] [--allow-no-migration] [--all-or-nothing [ALL-OR-NOTHING]] [--configuration CONFIGURATION] [--em EM] [--conn CONN] [--] [<version>]
```

When I run the update commands this happens:

```
 /var/www/mautic# sudo -u www-data php8.1 /var/www/mautic/bin/console mautic:update:find
Version 5.0.2 of Mautic is available for download. Please visit https://github.com/mautic/mautic/releases/tag/5.0.2 for more information.
To update, you can run 'php bin/console mautic:update:apply' from the command line.

/var/www/mautic# sudo -u www-data php8.1 /var/www/mautic/bin/console mautic:update:apply
Are you sure you wish to update Mautic to the latest version? y
Step    6 [--->------------------------] Removing deleted files

<warning>IMPORTANT: Run the same command again with --finish. For example 'php bin/console mautic:update:apply --finish'</warning>
root@MT-r1-GLT-EM:/var/www/mautic# sudo -u www-data php8.1 /var/www/mautic/bin/console mautic:update:apply --finish
Step    1 [>---------------------------] Migrating database schema...

An error occurred while updating the database. Check log for more details.
```

watching the log:
```
[2024-01-30 06:28:20] mautic.ERROR: Failed to dispatch a message to messenger. Handling "Mautic\MessengerBundle\Message\PageHitNotification" failed: An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'p0_.group_id' in 'field list' [] {"hostname":"mauticsrv","pid":91134}
[2024-01-30 06:28:20] mautic.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Exception\InvalidFieldNameException: "An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'p0_.group_id' in 'field list'" at /var/www/mautic/vendor/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php line 67 {"exception":"[object] (Doctrine\\DBAL\\Exception\\InvalidFieldNameException(code: 1054): An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'p0_.group_id' in 'field list' at /var/www/mautic/vendor/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php:67, Doctrine\\DBAL\\Driver\\PDO\\Exception(code: 1054): SQLSTATE[42S22]: Column not found: 1054 Unknown column 'p0_.group_id' in 'field list' at /var/www/mautic/vendor/doctrine/dbal/src/Driver/PDO/Exception.php:28, PDOException(code: 42S22): SQLSTATE[42S22]: Column not found: 1054 Unknown column 'p0_.group_id' in 'field list' at /var/www/mautic/vendor/doctrine/dbal/src/Driver/PDO/Statement.php:121)"} {"hostname":"mauticsrv","pid":91134}
[2024-01-30 06:28:20] mautic.CRITICAL: Exception thrown when handling an exception (InvalidArgumentException: The HTTP status code "1054" is not valid. at /var/www/mautic/vendor/symfony/http-foundation/Response.php line 470) {"exception":"[object] (InvalidArgumentException(code: 0): The HTTP status code \"1054\" is not valid. at /var/www/mautic/vendor/symfony/http-foundation/Response.php:470)"} {"hostname":"mauticsrv","pid":91134}
[2024-01-30 06:28:20] mautic.ERROR: InvalidArgumentException: The HTTP status code "1054" is not valid. - in file /var/www/mautic/vendor/symfony/http-foundation/Response.php - at line 470 [] {"hostname":"mauticsrv","pid":91134}
[2024-01-30 01:28:22] mautic.ERROR: Failed to dispatch a message to messenger. Handling "Mautic\MessengerBundle\Message\PageHitNotification" failed: An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'p0_.group_id' in 'field list' [] {"hostname":"mauticsrv","pid":91130}
[2024-01-30 01:28:22] mautic.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Exception\InvalidFieldNameException: "An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'p0_.group_id' in 'field list'" at /var/www/mautic/vendor/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php line 67 {"exception":"[object] (Doctrine\\DBAL\\Exception\\InvalidFieldNameException(code: 1054): An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'p0_.group_id' in 'field list' at /var/www/mautic/vendor/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php:67, Doctrine\\DBAL\\Driver\\PDO\\Exception(code: 1054): SQLSTATE[42S22]: Column not found: 1054 Unknown column 'p0_.group_id' in 'field list' at /var/www/mautic/vendor/doctrine/dbal/src/Driver/PDO/Exception.php:28, PDOException(code: 42S22): SQLSTATE[42S22]: Column not found: 1054 Unknown column 'p0_.group_id' in 'field list' at /var/www/mautic/vendor/doctrine/dbal/src/Driver/PDO/Statement.php:121)"} {"hostname":"mauticsrv","pid":91130}
[2024-01-30 01:28:22] mautic.CRITICAL: Exception thrown when handling an exception (InvalidArgumentException: The HTTP status code "1054" is not valid. at /var/www/mautic/vendor/symfony/http-foundation/Response.php line 470) {"exception":"[object] (InvalidArgumentException(code: 0): The HTTP status code \"1054\" is not valid. at /var/www/mautic/vendor/symfony/http-foundation/Response.php:470)"} {"hostname":"mauticsrv","pid":91130}
[2024-01-30 01:28:22] mautic.ERROR: InvalidArgumentException: The HTTP status code "1054" is not valid. - in file /var/www/mautic/vendor/symfony/http-foundation/Response.php - at line 470 [] {"hostname":"mauticsrv","pid":91130}
```

Any idea how to fix this or what I am doing wrong?
I tried clearing cache etc.

### How can we reproduce this issue?

When updating in console to mautic 5.x the apply --finish command fails at the database update.

I also tried:
/var/www/mautic# sudo -u www-data php8.1 /var/www/mautic/bin/console doctrine:migration:migrate

 WARNING! You are about to execute a migration in database "mautic" that could result in schema changes and data loss. Are you sure you wish to continue? (yes/no) [yes]:
 > yes

                                                                                                                        
 [WARNING] You have 49 previously executed migrations in the database that are not registered migrations.               
                                                                                                                        
Are you sure you wish to continue? (yes/no) [yes]:
 > yes

```
[notice] Migrating up to Mautic\Migrations\Versionzz20230929183000
[error] Migration Mautic\Migrations\Version20200415135706 failed during Pre-Checks. Error: "Unknown column type "json_array" requested. Any Doctrine type that you use has to be registered with \Doctrine\DBAL\Types\Type::addType(). You can get a list of all the known types with \Doctrine\DBAL\Types\Type::getTypesMap(). If this error occurs during database introspection then you might have forgotten to register all database types for a Doctrine Type. Use AbstractPlatform#registerDoctrineTypeMapping() or have your custom types implement Type#getMappedDatabaseTypes(). If the type name is empty you might have a problem with the cache or forgot some mapping information."

In Exception.php line 115:
                                                                                                                                                                                              
  Unknown column type "json_array" requested. Any Doctrine type that you use has to be registered with \Doctrine\DBAL\Types\Type::addType(). You can get a list of all the known types with   
  \Doctrine\DBAL\Types\Type::getTypesMap(). If this error occurs during database introspection then you might have forgotten to register all database types for a Doctrine Type. Use Abstrac  
  tPlatform#registerDoctrineTypeMapping() or have your custom types implement Type#getMappedDatabaseTypes(). If the type name is empty you might have a problem with the cache or forgot som  
  e mapping information.                                                                                                                                                                      
                                                                                                                                                                                              

doctrine:migrations:migrate [--write-sql [WRITE-SQL]] [--dry-run] [--query-time] [--allow-no-migration] [--all-or-nothing [ALL-OR-NOTHING]] [--configuration CONFIGURATION] [--em EM] [--conn CONN] [--] [<version>]
```

When I run the update commands this happens:

```
 /var/www/mautic# sudo -u www-data php8.1 /var/www/mautic/bin/console mautic:update:find
Version 5.0.2 of Mautic is available for download. Please visit https://github.com/mautic/mautic/releases/tag/5.0.2 for more information.
To update, you can run 'php bin/console mautic:update:apply' from the command line.

/var/www/mautic# sudo -u www-data php8.1 /var/www/mautic/bin/console mautic:update:apply
Are you sure you wish to update Mautic to the latest version? y
Step    6 [--->------------------------] Removing deleted files

<warning>IMPORTANT: Run the same command again with --finish. For example 'php bin/console mautic:update:apply --finish'</warning>
root@MT-r1-GLT-EM:/var/www/mautic# sudo -u www-data php8.1 /var/www/mautic/bin/console mautic:update:apply --finish
Step    1 [>---------------------------] Migrating database schema...

An error occurred while updating the database. Check log for more details.
```

watching the log:
```
[2024-01-30 06:28:20] mautic.ERROR: Failed to dispatch a message to messenger. Handling "Mautic\MessengerBundle\Message\PageHitNotification" failed: An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'p0_.group_id' in 'field list' [] {"hostname":"mauticsrv","pid":91134}
[2024-01-30 06:28:20] mautic.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Exception\InvalidFieldNameException: "An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'p0_.group_id' in 'field list'" at /var/www/mautic/vendor/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php line 67 {"exception":"[object] (Doctrine\\DBAL\\Exception\\InvalidFieldNameException(code: 1054): An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'p0_.group_id' in 'field list' at /var/www/mautic/vendor/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php:67, Doctrine\\DBAL\\Driver\\PDO\\Exception(code: 1054): SQLSTATE[42S22]: Column not found: 1054 Unknown column 'p0_.group_id' in 'field list' at /var/www/mautic/vendor/doctrine/dbal/src/Driver/PDO/Exception.php:28, PDOException(code: 42S22): SQLSTATE[42S22]: Column not found: 1054 Unknown column 'p0_.group_id' in 'field list' at /var/www/mautic/vendor/doctrine/dbal/src/Driver/PDO/Statement.php:121)"} {"hostname":"mauticsrv","pid":91134}
[2024-01-30 06:28:20] mautic.CRITICAL: Exception thrown when handling an exception (InvalidArgumentException: The HTTP status code "1054" is not valid. at /var/www/mautic/vendor/symfony/http-foundation/Response.php line 470) {"exception":"[object] (InvalidArgumentException(code: 0): The HTTP status code \"1054\" is not valid. at /var/www/mautic/vendor/symfony/http-foundation/Response.php:470)"} {"hostname":"mauticsrv","pid":91134}
[2024-01-30 06:28:20] mautic.ERROR: InvalidArgumentException: The HTTP status code "1054" is not valid. - in file /var/www/mautic/vendor/symfony/http-foundation/Response.php - at line 470 [] {"hostname":"mauticsrv","pid":91134}
[2024-01-30 01:28:22] mautic.ERROR: Failed to dispatch a message to messenger. Handling "Mautic\MessengerBundle\Message\PageHitNotification" failed: An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'p0_.group_id' in 'field list' [] {"hostname":"mauticsrv","pid":91130}
[2024-01-30 01:28:22] mautic.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Exception\InvalidFieldNameException: "An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'p0_.group_id' in 'field list'" at /var/www/mautic/vendor/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php line 67 {"exception":"[object] (Doctrine\\DBAL\\Exception\\InvalidFieldNameException(code: 1054): An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'p0_.group_id' in 'field list' at /var/www/mautic/vendor/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php:67, Doctrine\\DBAL\\Driver\\PDO\\Exception(code: 1054): SQLSTATE[42S22]: Column not found: 1054 Unknown column 'p0_.group_id' in 'field list' at /var/www/mautic/vendor/doctrine/dbal/src/Driver/PDO/Exception.php:28, PDOException(code: 42S22): SQLSTATE[42S22]: Column not found: 1054 Unknown column 'p0_.group_id' in 'field list' at /var/www/mautic/vendor/doctrine/dbal/src/Driver/PDO/Statement.php:121)"} {"hostname":"mauticsrv","pid":91130}
[2024-01-30 01:28:22] mautic.CRITICAL: Exception thrown when handling an exception (InvalidArgumentException: The HTTP status code "1054" is not valid. at /var/www/mautic/vendor/symfony/http-foundation/Response.php line 470) {"exception":"[object] (InvalidArgumentException(code: 0): The HTTP status code \"1054\" is not valid. at /var/www/mautic/vendor/symfony/http-foundation/Response.php:470)"} {"hostname":"mauticsrv","pid":91130}
[2024-01-30 01:28:22] mautic.ERROR: InvalidArgumentException: The HTTP status code "1054" is not valid. - in file /var/www/mautic/vendor/symfony/http-foundation/Response.php - at line 470 [] {"hostname":"mauticsrv","pid":91130}
```

Any idea how to fix this or what I am doing wrong?
I tried clearing cache etc.

### Relevant log output

```shell
[2024-01-30 06:28:20] mautic.ERROR: Failed to dispatch a message to messenger. Handling "Mautic\MessengerBundle\Message\PageHitNotification" failed: An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'p0_.group_id' in 'field list' [] {"hostname":"mauticsrv","pid":91134}
[2024-01-30 06:28:20] mautic.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Exception\InvalidFieldNameException: "An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'p0_.group_id' in 'field list'" at /var/www/mautic/vendor/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php line 67 {"exception":"[object] (Doctrine\\DBAL\\Exception\\InvalidFieldNameException(code: 1054): An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'p0_.group_id' in 'field list' at /var/www/mautic/vendor/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php:67, Doctrine\\DBAL\\Driver\\PDO\\Exception(code: 1054): SQLSTATE[42S22]: Column not found: 1054 Unknown column 'p0_.group_id' in 'field list' at /var/www/mautic/vendor/doctrine/dbal/src/Driver/PDO/Exception.php:28, PDOException(code: 42S22): SQLSTATE[42S22]: Column not found: 1054 Unknown column 'p0_.group_id' in 'field list' at /var/www/mautic/vendor/doctrine/dbal/src/Driver/PDO/Statement.php:121)"} {"hostname":"mauticsrv","pid":91134}
[2024-01-30 06:28:20] mautic.CRITICAL: Exception thrown when handling an exception (InvalidArgumentException: The HTTP status code "1054" is not valid. at /var/www/mautic/vendor/symfony/http-foundation/Response.php line 470) {"exception":"[object] (InvalidArgumentException(code: 0): The HTTP status code \"1054\" is not valid. at /var/www/mautic/vendor/symfony/http-foundation/Response.php:470)"} {"hostname":"mauticsrv","pid":91134}
[2024-01-30 06:28:20] mautic.ERROR: InvalidArgumentException: The HTTP status code "1054" is not valid. - in file /var/www/mautic/vendor/symfony/http-foundation/Response.php - at line 470 [] {"hostname":"mauticsrv","pid":91134}
[2024-01-30 01:28:22] mautic.ERROR: Failed to dispatch a message to messenger. Handling "Mautic\MessengerBundle\Message\PageHitNotification" failed: An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'p0_.group_id' in 'field list' [] {"hostname":"mauticsrv","pid":91130}
[2024-01-30 01:28:22] mautic.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Exception\InvalidFieldNameException: "An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'p0_.group_id' in 'field list'" at /var/www/mautic/vendor/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php line 67 {"exception":"[object] (Doctrine\\DBAL\\Exception\\InvalidFieldNameException(code: 1054): An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'p0_.group_id' in 'field list' at /var/www/mautic/vendor/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php:67, Doctrine\\DBAL\\Driver\\PDO\\Exception(code: 1054): SQLSTATE[42S22]: Column not found: 1054 Unknown column 'p0_.group_id' in 'field list' at /var/www/mautic/vendor/doctrine/dbal/src/Driver/PDO/Exception.php:28, PDOException(code: 42S22): SQLSTATE[42S22]: Column not found: 1054 Unknown column 'p0_.group_id' in 'field list' at /var/www/mautic/vendor/doctrine/dbal/src/Driver/PDO/Statement.php:121)"} {"hostname":"mauticsrv","pid":91130}
[2024-01-30 01:28:22] mautic.CRITICAL: Exception thrown when handling an exception (InvalidArgumentException: The HTTP status code "1054" is not valid. at /var/www/mautic/vendor/symfony/http-foundation/Response.php line 470) {"exception":"[object] (InvalidArgumentException(code: 0): The HTTP status code \"1054\" is not valid. at /var/www/mautic/vendor/symfony/http-foundation/Response.php:470)"} {"hostname":"mauticsrv","pid":91130}
[2024-01-30 01:28:22] mautic.ERROR: InvalidArgumentException: The HTTP status code "1054" is not valid. - in file /var/www/mautic/vendor/symfony/http-foundation/Response.php - at line 470 [] {"hostname":"mauticsrv","pid":91130}
```


### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

<br /><hr>
Care about this issue? Want to get it resolved sooner? If you are a <a href='https://www.mautic.org/become-a-member-of-mautic'>member of Mautic</a>, you can add some funds to the <a href='https://opencollective.com/mautic/projects/bounties'>Bounties Project</a> so that the person who completes this task can claim those funds once it is merged by a member of the core team! Read the docs <a href='https://contribute.mautic.org/product-team/mautic-bounty-programme'>here.</a>

**Comment by openarun** [comment]
+1

**Comment by openarun** [comment]
`[notice] Migrating up to Mautic\Migrations\Versionzz20230929183000
[error] Migration Mautic\Migrations\Version20200415135706 failed during Pre-Checks. Error: "Unknown column type "json_array" requested. Any Doctrine type that you use has to be registered with \Doctrine\DBAL\Types\Type::addType(). You can get a list of all the known types with \Doctrine\DBAL\Types\Type::getTypesMap(). If this error occurs during database introspection then you might have forgotten to register all database types for a Doctrine Type. Use AbstractPlatform#registerDoctrineTypeMapping() or have your custom types implement Type#getMappedDatabaseTypes(). If the type name is empty you might have a problem with the cache or forgot some mapping information."        
`

**Comment by escopecz** [comment]
@dadarya0 had the same issue with the `"Unknown column type "json_array"` yesterday and solved it with "I have deleted my whole db after that it is fixed" but that was on a development environment so he was OK with losing the data. That's not an option on production. We weren't able to search for `json_array` in the Mautic 5 codebase so it must be in the database. Could you export the database to raw SQL and search for `json_array` in there? That way we'll be able to locate the problem.

**Comment by openarun** [comment]
@escopecz ; I took a dump and grep-ed through the db, I found the following occurrences of `json_array` in the tables below:

[bak_reports]
  `settings` json DEFAULT NULL COMMENT '(DC2Type:json_array)',

[bak_sms_message_stats]
  `details` json NOT NULL COMMENT '(DC2Type:json_array)',

[bak_sync_object_mapping]
  `internal_storage` json NOT NULL COMMENT '(DC2Type:json_array)',

[bak_tweet_stats]
  `response_details` json DEFAULT NULL COMMENT '(DC2Type:json_array)',

Hope that helps to resolve this issue or create a quick patch. :)

**Comment by openarun** [comment]
I managed to manually fix the above issue by replacing `(DC2Type:json_array)` with `(DC2Type:json)` in the database.

But now, encountered another migration error:

```
[notice] Migrating up to Mautic\Migrations\Versionzz20230929183000
[notice] Migration Mautic\Migrations\Version20221128145933 skipped during Pre-Checks. Reason: "Schema includes this migration"
[error] Migration Mautic\Migrations\Version20230131133732 failed during Execution. Error: "An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'last_built_date' in 'lead_lists'"
```

**Update : I fixed the unknown column `last_built_date` issue by adding a new column with DATETIME type and nullable, which finished the migration :)**

**Comment by escopecz** [comment]
This is weird. How come you were missing the last_built_date in Mautic 4.4.11 when it was [added in Mautic 4.2.0](https://github.com/mautic/mautic/pull/10772)?

We'll have to write a migration for the deprecated JSON_ARRAY type that were changed in PRs like https://github.com/mautic/mautic/pull/12288 but the migrations weren't added. I suppose no one expected that Doctrine would fail just because the comments do not match.

I've marked this ticket as a regression to add the migration ASAP.

**Comment by openarun** [comment]
Also, I found another issue when opening a contact view, got a 500 error. The table `lead_tags` had a missing column `description` .
Might need a migration for that too.

**Comment by escopecz** [comment]
@openarun are you sure you've run all the migrations for M4 before you've upgraded to M5? As in my previous comment, you were missing a column from Mautic 4.2.0. I can't imagine how could your Mautic 4.4.11 worked without that.

**Comment by wsarisslz** [comment]
I tried everything and I can’t get around that json_array error… 

I did execute all the migration commands.

**Comment by openarun** [comment]
> @openarun are you sure you've run all the migrations for M4 before you've upgraded to M5? As in my previous comment, you were missing a column from Mautic 4.2.0. I can't imagine how could your Mautic 4.4.11 worked without that.

Yes, it was working fine without any issue in 4.4.11; and yes no manual change was made to the DB.

**Comment by mallezie** [comment]
Ok, so i found the problem. 

Problem is 'json_array' is deprecated in doctrine/dbal 2.x and removed in doctrine/dbal 3.x. 
M4 contains types for json_array, and the migration to update those is only in M5. But that is too late, since then the datatype is already gone, and it fails before it can run the migration fixing it. 

To fix this, IMO we should need M4.x release with following migration. https://github.com/mautic/mautic/pull/12491/files#diff-1810a235c9145afb437c52cb1e48d72d0c99c5bb0d1047e10bb798cef00acef6

That way we fix the datatype before the update to M5. 

I'm still testing some stuff, but what strikes me the most is that this issue is not encountered more. When installing last mautic 4 (ddev install) i see this 'deprecated / removed' type being installed. Causing that actually a new last M4 installed is not updatable. 

I'll provide more info and possible fix PR's tomorrow.

**Comment by mallezie** [comment]
Ok, forget above comment, the migration indeed runs first, and should fix that. 

If you still encounter this, it is because in some extra tables there is still a json_array left. In case of issue poster. 

> [bak_reports]
> settings json DEFAULT NULL COMMENT '(DC2Type:json_array)',
> 
> [bak_sms_message_stats]
> details json NOT NULL COMMENT '(DC2Type:json_array)',
> 
> [bak_sync_object_mapping]
> internal_storage json NOT NULL COMMENT '(DC2Type:json_array)',
> 
> [bak_tweet_stats]
> response_details json DEFAULT NULL COMMENT '(DC2Type:json_array)',

This is in backup tables, probably becuase of a reinstall. I'm not sure this is in scope for Mautic to fix. You can remove the tables, and then it will continue. 

Where we encountered the same issue is in the custom_field table. This table only get's created AFAICT when you have custom fields. Not in a default install. 

```
MariaDB [db]> show full columns from custom_field;
+-----------------------------------+------------------+--------------------+------+-----+---------+----------------+---------------------------------+----------------------+
| Field                             | Type             | Collation          | Null | Key | Default | Extra          | Privileges                      | Comment              |
+-----------------------------------+------------------+--------------------+------+-----+---------+----------------+---------------------------------+----------------------+
| id                                | int(10) unsigned | NULL               | NO   | PRI | NULL    | auto_increment | select,insert,update,references |                      |
| custom_object_id                  | int(10) unsigned | NULL               | NO   | MUL | NULL    |                | select,insert,update,references |                      |
| is_published                      | tinyint(1)       | NULL               | NO   |     | NULL    |                | select,insert,update,references |                      |
| date_added                        | datetime         | NULL               | YES  |     | NULL    |                | select,insert,update,references |                      |
| created_by                        | int(11)          | NULL               | YES  |     | NULL    |                | select,insert,update,references |                      |
| created_by_user                   | varchar(191)     | utf8mb4_unicode_ci | YES  |     | NULL    |                | select,insert,update,references |                      |
| date_modified                     | datetime         | NULL               | YES  |     | NULL    |                | select,insert,update,references |                      |
| modified_by                       | int(11)          | NULL               | YES  |     | NULL    |                | select,insert,update,references |                      |
| modified_by_user                  | varchar(191)     | utf8mb4_unicode_ci | YES  |     | NULL    |                | select,insert,update,references |                      |
| checked_out                       | datetime         | NULL               | YES  |     | NULL    |                | select,insert,update,references |                      |
| checked_out_by                    | int(11)          | NULL               | YES  |     | NULL    |                | select,insert,update,references |                      |
| checked_out_by_user               | varchar(191)     | utf8mb4_unicode_ci | YES  |     | NULL    |                | select,insert,update,references |                      |
| label                             | varchar(191)     | utf8mb4_unicode_ci | NO   |     | NULL    |                | select,insert,update,references |                      |
| alias                             | varchar(191)     | utf8mb4_unicode_ci | NO   | MUL | NULL    |                | select,insert,update,references |                      |
| type                              | varchar(191)     | utf8mb4_unicode_ci | NO   |     | NULL    |                | select,insert,update,references |                      |
| field_order                       | int(11)          | NULL               | YES  |     | NULL    |                | select,insert,update,references |                      |
| required                          | tinyint(1)       | NULL               | NO   |     | 0       |                | select,insert,update,references |                      |
| default_value                     | varchar(191)     | utf8mb4_unicode_ci | YES  |     | NULL    |                | select,insert,update,references |                      |
| params                            | longtext         | utf8mb4_bin        | YES  |     | NULL    |                | select,insert,update,references | (DC2Type:json_array) |
| show_in_custom_object_detail_list | tinyint(1)       | NULL               | NO   |     | 1       |                | select,insert,update,references |                      |
| show_in_contact_detail_list       | tinyint(1)       | NULL               | NO   |     | 1       |                | select,insert,update,references |                      |
| uuid                              | char(36)         | utf8mb4_unicode_ci | YES  |     | NULL    |                | select,insert,update,references | (DC2Type:guid)       |
+-----------------------------------+------------------+--------------------+------+-----+---------+----------------+---------------------------------+----------------------+
```

**Comment by mallezie** [comment]
Ok, the custom_field table comes from https://github.com/acquia/mc-cs-plugin-custom-objects
So nothing to do in mautic itself IMO.

**Comment by wsarisslz** [comment]
Hi @mallezie thanks for all the work.
Yes I was using custom objects. So I guess its not Mautic 5 compatible. Damn.

I can't believe there is no ecommerce custom object solution for mautic / shopify available. 
That would be a hit.

**Comment by dandumit** [comment]
Hello all,
IS there any fix that I can use to restore functionality  ?  I have tried to upgrade from version 4.  What can I do ?

Daniel

**Comment by Gregy1** [comment]
Hi @escopecz ...if you could take a look:

I've just migrated from 4.4.11 to 5.0.4 and think the similar problem is with my instance.

Log keeps showing this error every time I click on segments tab (even unable to click and open single segment or change the date format even that my date/time format is set to "j F Y H:i"):

[2024-04-30T20:40:47.326160+00:00] mautic.CRITICAL: Exception thrown when handling an exception (InvalidArgumentException: The HTTP status code "1176" is not valid. at /home/zhpplowbkq/domains/mtest.leasing.sos.pl/public_html/vendor/symfony/http-foundation/Response.php line 470) {"exception":"[object] (InvalidArgumentException(code: 0): The HTTP status code \"1176\" is not valid. at /home/zhpplowbkq/domains/mtest.leasing.sos.pl/public_html/vendor/symfony/http-foundation/Response.php:470)"} {"hostname":"s167.cyber-folks.pl","pid":3511}
[2024-04-30T20:40:47.326214+00:00] mautic.CRITICAL: Uncaught PHP Exception InvalidArgumentException: "The HTTP status code "1176" is not valid." at /home/zhpplowbkq/domains/mtest.leasing.sos.pl/public_html/vendor/symfony/http-foundation/Response.php line 470 {"exception":"[object] (InvalidArgumentException(code: 0): The HTTP status code \"1176\" is not valid. at /home/zhpplowbkq/domains/mtest.leasing.sos.pl/public_html/vendor/symfony/http-foundation/Response.php:470)\n[previous exception] [object] (Doctrine\\DBAL\\Exception\\DriverException(code: 1176): An exception occurred while executing a query: SQLSTATE[42000]: Syntax error or access violation: 1176 Key 'koch_manually_removed' doesn't exist in table 'l' at /home/zhpplowbkq/domains/mtest.leasing.sos.pl/public_html/vendor/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php:117)\n[previous exception] [object] (Doctrine\\DBAL\\Driver\\PDO\\Exception(code: 1176): SQLSTATE[42000]: Syntax error or access violation: 1176 Key 'koch_manually_removed' doesn't exist in table 'l' at /home/zhpplowbkq/domains/mtest.leasing.sos.pl/public_html/vendor/doctrine/dbal/src/Driver/PDO/Exception.php:28)\n[previous exception] [object] (PDOException(code: 42000): SQLSTATE[42000]: Syntax error or access violation: 1176 Key 'koch_manually_removed' doesn't exist in table 'l' at /home/zhpplowbkq/domains/mtest.leasing.sos.pl/public_html/vendor/doctrine/dbal/src/Driver/PDO/Statement.php:130)"} {"hostname":"s167.cyber-folks.pl","pid":3511}

Is there a hope to solve this problem in general? (at least I have backup:)

Kind regards,

Greg

**Comment by escopecz** [comment]
@Gregy1 Notice `'koch_manually_removed' doesn't exist in table 'l'` in the error message. It means that the table is missing one column that the code is expecting. It seems that your migrations did not finish successfully. Please run `bin/console doctrine:migrations:migrate` and see if it all passes.

**Comment by pstevens71** [comment]
If memory serves me correctly, one of the people above mentioned that this json array was "defunct" ie: added previously and then removed in a subsequent migration.  I think what I did was a grep to look for the docterine where it was added and then removed. and deleted both from the docterine files, then it got past this problem on the install.

**Comment by trianity** [comment]
@escopecz  @RCheesley 
I also ran into this problem during a Mautic 4.4.12 -> 5.1.1 upgrade, but I noticed that the database contained a backup version, and the "json_array" column comments were only in the old tables that were not in use.
After deleting the old, unused tables, the migration ran without error.
You can check the database if any deprecated column comments found:

`SELECT table_name, column_name, column_comment
FROM information_schema.columns
WHERE table_schema = 'your_database_name'
  AND column_comment LIKE '%json_array%';
`

**Comment by RCheesley** [comment]
Thanks @trianity for that additional information, I'm sure it'll be helpful for folks coming across this issue in the future!

Given this seems to be either a case of users have backed up tables (which won't get upgraded) or having custom fields from a third party plugin, I'm going to go ahead and close this issue accordingly.
